spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/features/TaskConfiguration.java

```diff
@@ -182,12 +183,14 @@ public TaskExecutionService taskService(LauncherRepository launcherRepository,
 			TaskAppDeploymentRequestCreator taskAppDeploymentRequestCreator,
 			TaskExplorer taskExplorer,
 			DataflowTaskExecutionDao dataflowTaskExecutionDao,
-			DataflowTaskExecutionMetadataDao dataflowTaskExecutionMetadataDao) {
+			DataflowTaskExecutionMetadataDao dataflowTaskExecutionMetadataDao,
+			OAuth2TokenUtilsService oauth2TokenUtilsService) {
```

Comment:
Instead of having this as a required dependency and bringing in `DummyOAuth2TokenUtilsService` when OAuth2 is disabled, we can have this as a non-required dependency in `DefaultTaskExecutionService` only when OAuth2 is enabled.

Reply:
Made the change